models/publication.go

```diff
@@ -1,10 +1,85 @@
 package models
 
-import "time"
+import (
+	"errors"
+	"strconv"
+	"time"
+)
 
 type PublicationId int64
 
 type Publication struct {
 	AuthorId     UserId    `json:"authorId"`
 	CreationTime time.Time `json:"creationTime"`
 }
+
+var NoNotesToPublishError = errors.New("There are no unpublished notes to publish")
+
+func (db *DB) PublishNotes(userId UserId) error {
+
+	myUnpublishedNotes, err := db.GetMyUnpublishedNotes(userId)
+	if err != nil {
+		return err
+	}
+
+	if len(myUnpublishedNotes) == 0 {
+		return NoNotesToPublishError
+	}
+
+	publicationId, err := db.StoreNewPublication(&Publication{AuthorId: userId, CreationTime: time.Now().UTC()})
+
+	sqlQuery := `
+		INSERT INTO note_to_publication_relationship (publication_id, note_id) 
+		VALUES ($1, $2)`
+
+	noteIds := make([]int64, len(myUnpublishedNotes))
+
+	{
+		i := 0
+		for noteId := range myUnpublishedNotes {
+			noteIds[i] = int64(noteId)
+			i++
+		}
+	}
+
+	values := make([]interface{}, len(noteIds)*2, len(noteIds)*2)
+	{
+		values[0] = publicationId
+		values[1] = noteIds[0]
+		for index, noteId := range noteIds {
+			if index == 0 {
+				continue
+			}
+			sqlQuery += ", ($" + strconv.Itoa(2*index+1) + ", $" + strconv.Itoa((2*index)+2) + ")"
```

Comment:
I think it's time to bring in an orm. 
1. This query is hard to read
2. [This](https://github.com/zingales/cerealnotes/pull/4/files/5492b12633d3d38ff0bba5e9180436bd92e891cd#diff-9a13b2d5511d854d164540d80e487c4aR64) query is hard to read (this one might be mostly because i'm bad at sql)
3. We talked about how auto creating databases for tests is hard right now

orm can help with all of that.